src/main/kotlin/io/github/soat7/myburguercontrol/infrastructure/persistence/item/entity/ItemEntity.kt

```diff
@@ -0,0 +1,39 @@
+package io.github.soat7.myburguercontrol.infrastructure.persistence.item.entity
+
+import jakarta.persistence.Column
+import jakarta.persistence.Entity
+import jakarta.persistence.GeneratedValue
+import jakarta.persistence.GenerationType
+import jakarta.persistence.Id
+import jakarta.persistence.Table
+import java.math.BigDecimal
+import java.time.Instant
+import java.util.UUID
+
+@Entity
+@Table(
+    name = "item",
+    schema = "myburguer"
+)
+data class ItemEntity(
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    @Column(name = "id", nullable = false)
+    val id: UUID? = null,
+
+    @Column(name = "description", nullable = false, length = 255)
+    val description: String,
+
+    @Column(name = "type", nullable = false, length = 255)
+    val type: String,
```

Comment:
Pode usar enumaration no hibernate facilita muito o mapper (tem como falar pra coluna ser uma string) https://www.baeldung.com/jpa-persisting-enums-in-jpa

Se voce quiser MUITO escrever um converter recomendo usar essa tecnica
https://www.baeldung.com/jpa-persisting-enums-in-jpa#converter

Reply:
Eu sei que podemos user coluna como Enumerated. Mas não vi ganho algum nisso já que temos uma classe de domain que contém o ENUM. Do jeito que fiz não precisamos ficar convertendo o ENUM de Domain para o ENUM de persistence

Reply:
Entendi..... É que eu sou adpeto de Enum ser global justamente pra poder usar o poder dele de ser mais assertivo em comparações, realmente ter um Enum pra cada camada é ruim. Não acha razoável centralizar esse enum?